### DISCUSSION TITLE
Add Query Parameters to Collections, Fields, and Relations API

### DISCUSSION THREAD

**Discussion by jofmi** [directus/directus#20183] [Feature Requests]
### Summary

Most Directus API Endpoint operations can be manipulated with global query parameters: https://docs.directus.io/reference/query.html

This would also be very useful for the Collections, Fields, and Relations endpoints.

### Motivation

This will be useful to assess the database structure via the API.

### Detailed Design

Add the same logic as is e.g. implemented for items: https://docs.directus.io/reference/items.html#query-parameters

### Must Have:
- API Implementation
- SDK Implementation

### Drawbacks

None that I can think of.

### Alternatives

None that I can think of.

### Adoption Strategy

No breaking change.

**Comment by jacobwise** [comment]
I think I'm trying to do something similar. I have a M2M between `directus_users` and `classes` called `favorites`. I want to get context if the current user has favorited a class. I've got a work around where I fetch all of the users who have favorited a class and just check if the current user is in that list but it would be nice if I could apply filtering rules to the fields object to accomplish that. I'm wondering if there is another way to do this or think about this?

**Reply by jacobwise** [reply]
I was able to get most of the way there using [deep](https://docs.directus.io/reference/query.html#deep).

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
